code-for-canada/project-thundercat
# [Accessibility] Adding dynamic password validation

# Description

This PR is introducing the dynamic password validation. Instead of always displaying all the password requirements, it analyses the user's password and displays only the missing requirements.

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

**Empty field:**

![image](https://user-images.githubusercontent.com/23021242/58975994-64596300-8794-11e9-9d4e-9aa36b2095be.png)

**_mypassword_ in password field:**

![image](https://user-images.githubusercontent.com/23021242/58976053-8eab2080-8794-11e9-99e6-0733b56df5de.png)

**_Mypassword_ in password field:**

![image](https://user-images.githubusercontent.com/23021242/58976074-9f5b9680-8794-11e9-82d3-0222ea6af02c.png)

**_Mypassword01_ in password field:**

![image](https://user-images.githubusercontent.com/23021242/58976113-b26e6680-8794-11e9-8ec6-cef73588e29e.png)

**_Mypassword01$_ in password field:**

![image](https://user-images.githubusercontent.com/23021242/58976161-ce720800-8794-11e9-8512-35401315aa8f.png)

# Testing

Manual steps to reproduce this functionality:

1.  Open the registration form in home page.
2.  Try different valid and invalid passwords and make sure that you get the expected results.

# Checklist

Applicable for all code changes.
